src/rgw/rgw_op.cc

```diff
@@ -590,7 +590,14 @@ void rgw_add_to_iam_environment(rgw::IAM::Environment& e, std::string&& key, std
 }
 
 void rgw_add_to_iam_environment(rgw::IAM::Environment& e, const std::string& key, const std::string& val){
-  e[key] = val;
+  if (!val.empty())
+    e[key] = val;
+}
+
+
+void rgw_add_to_iam_environment(rgw::IAM::Environment& e, const char* key, const char* val){
```

Comment:
I'd be tempted to merge these again and just use boost::string_view

Reply:
Now the project is C++17, so there is std::string_view...